assets/js/googlesitekit/datastore/site/html.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * core/site data store: HTML for URL.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * WordPress dependencies
+ */
+import { isURL, addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const { createRegistryControl } = Data;
+
+const fetchHTMLForURLStore = createFetchStore( {
+	baseName: 'getHTMLForURL',
+	argsToParams: ( url ) => {
+		invariant( isURL( url ), 'a valid url is required to fetch HTML.' );
+		return { url };
+	},
+	controlCallback: async ( { url } ) => {
+		const fetchHTMLOptions = {
+			credentials: 'omit',
+		};
+		const fetchHTMLQueryArgs = {
+			// Indicates a tag checking request. This lets Site Kit know not to output its own tags.
+			tagverify: 1,
+			// Add a timestamp for cache-busting.
+			timestamp: Date.now(),
+		};
+		const response = await fetch( addQueryArgs( url, fetchHTMLQueryArgs ), fetchHTMLOptions );
+
+		// If response contains HTML, return that. Return null in other cases.
+		try {
+			const html = await response.text();
+			return html !== undefined ? html : null;
+		} catch {
+			return null;
+		}
+	},
+	reducerCallback: ( state, htmlForURL, { url } ) => {
+		return {
+			...state,
+			htmlForURL: {
+				...state.htmlForURL,
+				[ url ]: htmlForURL,
+			},
+		};
+	},
+} );
+
+// Actions
+const RESET_HTML_FOR_URL = 'RESET_HTML_FOR_URL';
+const WAIT_FOR_HTML_FOR_URL = 'WAIT_FOR_HTML_FOR_URL';
+
+export const BASE_INITIAL_STATE = {
+	htmlForURL: {},
+};
+
+const baseActions = {
+	/**
+	 * Resets the HTML for a given URL.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} url URL for which the HTML should be reset.
+	 * @return {Object} Redux-style action.
+	 */
+	*resetHTMLForURL( url ) {
```

Comment:
Should we maybe call this "clear" or something like that? "Reset" implies an initial value of something other than `null` or `undefined` to me. Just a thought.